core/forkid/forkid.go

```diff
@@ -67,17 +67,17 @@ type Filter func(id ID) error
 
 // NewID calculates the Ethereum fork ID from the chain config and head.
 func NewID(chain Blockchain) ID {
-	return newID(
+	return NewStaticID(
```

Comment:
I think we should improve this by changing the callers of NewID across the codebase to call it with the config, genesis and current block instead.